crates/broker/src/provers/default.rs

```diff
@@ -0,0 +1,490 @@
+// Copyright (c) 2025 RISC Zero, Inc.
+//
+// All rights reserved.
+
+use std::{borrow::Borrow, collections::HashMap, sync::Arc};
+
+use crate::config::ProverConf;
+use crate::provers::{ExecutorResp, ProofResult, Prover, ProverError};
+use anyhow::{Context, Result as AnyhowResult};
+use async_trait::async_trait;
+use risc0_zkvm::{
+    default_executor, default_prover, ExecutorEnv, ProveInfo, ProverOpts, Receipt, SessionInfo,
+    VERSION,
+};
+use tokio::sync::Mutex;
+use uuid::Uuid;
+
+#[derive(Debug, Default)]
+pub struct DefaultProver {
+    state: Arc<ProverState>,
+}
+
+#[derive(Debug, Default)]
+struct ProverState {
+    inputs: Mutex<HashMap<String, Vec<u8>>>,
+    images: Mutex<HashMap<String, Vec<u8>>>,
+    proofs: Mutex<HashMap<String, ProofData>>,
```

Comment:
Is `Mutex` required over a `RwLock`? Assume some tests will have some parallelism and just wondering if can avoid lock contention. No big deal, just curious.

Reply:
I am not sure if this is really critical, but I changed it to `RwLock`